src/client/dfuse/SConscript

```diff
@@ -0,0 +1,99 @@
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions, and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the following disclaimer in the
+#    documentation and/or materials provided with the distribution.
+#
+# 3. In addition, redistributions of modified forms of the source or binary
+#    code must carry prominent notices stating that the original code was
+#    changed and the date of the change.
+#
+#  4. All publications or advertising materials mentioning features or use of
+#     this software are asked, but not required, to acknowledge that it was
+#     developed by Intel Corporation and credit the contributors.
+#
+# 5. Neither the name of Intel Corporation, nor the name of any Contributor
+#    may be used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER BE LIABLE FOR ANY
+# DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+# THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""Build iof components"""
+
+import os
+import sys
+
+IOF_VERSION = "0.0.1"
+
+# Desired compiler flags that will be used if the compiler supports them
+DESIRED_FLAGS = ['-Wno-missing-braces',
+                 '-Wno-ignored-attributes',
+                 '-Wunknown-warning-option',
+                 '-Wno-gnu-designator',
+                 '-Wshadow',
+                 '-Wno-gnu-zero-variadic-macro-arguments']
+
+def run_checks(env):
+    """Run all configure time checks"""
+
+    cenv = env.Clone()
+    cenv.Append(CFLAGS='-Werror')
+    config = Configure(cenv)
+
+    cmd = 'setfattr'
+
+    if not config.CheckProg(cmd):
+        print('%s command not installed, extended attribute test ' \
+              'will not work' % cmd)
+
+    if config.CheckHeader('stdatomic.h'):
+        env.AppendUnique(CPPDEFINES=['HAVE_STDATOMIC=1'])
+
+    config.Finish()
+
+    env.AppendIfSupported(CFLAGS=DESIRED_FLAGS)
+
+    print('Compiler options: %s %s' % (env.get('CC'),
+                                       ' '.join(env.get('CFLAGS'))))
+
+def scons():
+    """Scons function"""
+
+    Import('env', 'prereqs')
+
+    VariantDir('build', '.', duplicate=0)
+
+    prereqs.require(["cart", "fuse", "ompi"])
```

Comment:
why do we need ompi?
we shouldn't require any MPI implementation on the client side other than for tests.

Reply:
IOF needed OMPI for testing only, and then all that it needed was orterun to launch the tests.